molgenis-data/src/test/java/org/molgenis/data/meta/model/AttributeTest.java

```diff
@@ -149,6 +312,24 @@ public void testIdValidationExpression() {
   // Regression test for https://github.com/molgenis/molgenis/issues/6566
   @Test
   public void testNewInstance() {
+    EntityType entityType = mock(EntityType.class);
+    Attribute typeAttr = when(mock(Attribute.class).getDataType()).thenReturn(STRING).getMock();
+    Attribute isNullableAttr = when(mock(Attribute.class).getDataType()).thenReturn(BOOL).getMock();
+    Attribute isAutoAttr = when(mock(Attribute.class).getDataType()).thenReturn(BOOL).getMock();
+    Attribute isVisibleAttr = when(mock(Attribute.class).getDataType()).thenReturn(BOOL).getMock();
+    Attribute isAggregatableAttr =
+        when(mock(Attribute.class).getDataType()).thenReturn(BOOL).getMock();
+    Attribute isReadOnlyAttr = when(mock(Attribute.class).getDataType()).thenReturn(BOOL).getMock();
+    Attribute isUniqueAttr = when(mock(Attribute.class).getDataType()).thenReturn(BOOL).getMock();
+    doReturn(typeAttr).when(entityType).getAttribute(TYPE);
+    doReturn(isNullableAttr).when(entityType).getAttribute(IS_NULLABLE);
+    doReturn(isAutoAttr).when(entityType).getAttribute(IS_AUTO);
+    doReturn(isVisibleAttr).when(entityType).getAttribute(IS_VISIBLE);
+    doReturn(isAggregatableAttr).when(entityType).getAttribute(IS_AGGREGATABLE);
+    doReturn(isReadOnlyAttr).when(entityType).getAttribute(IS_READ_ONLY);
+    doReturn(isUniqueAttr).when(entityType).getAttribute(IS_UNIQUE);
```

Comment:
Is it possible to extract a chunk of these lines? They're repeated quite a few times in this class.

Reply:
maybe, but strict mocking makes it difficult, I find it far more readable is they are at least in one place.

Reply:
Word.